containers/jetty/src/main/java/com/yahoo/athenz/container/log/JettyConnectionLogger/JettyConnectionLogger.java

```diff
@@ -0,0 +1,284 @@
+        handleListenerInvocation("Connection.Listener", "onOpened", "%h", Stream.of(connection).collect(Collectors.toList()), () -> {
+            SocketChannelEndPoint endpoint = findUnderlyingSocketEndpoint(connection.getEndPoint());
+            ConnectionInfo info = connectionInfo.get(endpoint);
+            if (info == null) {
+                info = ConnectionInfo.from(endpoint);
+                connectionInfo.put(endpoint, info);
+            }
+            if (connection instanceof SslConnection) {
+                SSLEngine sslEngine = ((SslConnection) connection).getSSLEngine();
+                sslToConnectionInfo.put(sslEngine, info);
+            }
+            if (connection.getEndPoint() instanceof ProxyConnectionFactory.ProxyEndPoint) {
+                InetSocketAddress remoteAddress = connection.getEndPoint().getRemoteAddress();
+                info.setRemoteAddress(remoteAddress);
+            }
+        });
+    }
+
+    @Override
+    public void onClosed(Connection connection) {
+        handleListenerInvocation("Connection.Listener", "onClosed", "%h", Stream.of(connection).collect(Collectors.toList()), () -> {
+            SocketChannelEndPoint endpoint = findUnderlyingSocketEndpoint(connection.getEndPoint());
+            ConnectionInfo info = connectionInfo.get(endpoint);
+            if (info == null) {
+                return; // Closed connection already handled
+            }
+            if (!endpoint.isOpen()) {
+                info.setClosedAt(System.currentTimeMillis());
+                // Only print / report failed connections
+                if (info.sslHandshakeFailureException != null) {
+                    connectionLog.log(info.toLogEntry());
+                    metric.increment(METRIC_NAME, info.toMetric());
+                }
+                connectionInfo.remove(endpoint);
+            }
+        });
+    }
+    //
+    // Connection.Listener methods end
+    //
+
+    //
+    // SslHandshakeListener methods start
+    //
+    @Override
+    public void handshakeSucceeded(Event event) {
+        SSLEngine sslEngine = event.getSSLEngine();
+        handleListenerInvocation("SslHandshakeListener", "handshakeSucceeded", "sslEngine=%h", Stream.of(sslEngine).collect(Collectors.toList()), () -> {
+            sslToConnectionInfo.remove(sslEngine);
+        });
+    }
+
+    @Override
+    public void handshakeFailed(Event event, Throwable failure) {
+        SSLEngine sslEngine = event.getSSLEngine();
+        handleListenerInvocation("SslHandshakeListener", "handshakeFailed", "sslEngine=%h,failure=%s", Stream.of(sslEngine, failure).collect(Collectors.toList()), () -> {
+            LOGGER.error(failure.getMessage());
+            ConnectionInfo info = sslToConnectionInfo.remove(sslEngine);
+            info.setSslHandshakeFailure(failure);
+        });
+    }
+    //
+    // SslHandshakeListener methods end
+    //
+
+    private void handleListenerInvocation(
```

Comment:
This method wraps the other "event listeners" in this class and writes the event to the main log (server.log) in debug mode. For example "DEBUG c.y.a.c.l.J.JettyConnectionLogger - Connection.Listener.onClosed(597fea07)".
We might want to consider removing it in the future if it will generate too much "noise".